DSharpPlus/Entities/Guild/DiscordGuild.cs

```diff
@@ -1795,6 +1796,43 @@ public async IAsyncEnumerable<DiscordMember> GetAllMembersAsync
         }
     }
 
+    /// <summary>
+    /// Requests that Discord send a list of guild members based on the specified arguments. This method will fire the GuildMembersChunked event.
+    /// <para>If no arguments aside from <paramref name="presences"/> and <paramref name="nonce"/> are specified, this will request all guild members.</para>
+    /// </summary>
+    /// <param name="query">Filters the returned members based on what the username starts with. Either this or <paramref name="userIds"/> must not be null.
+    /// The <paramref name="limit"/> must also be greater than 0 if this is specified.</param>
+    /// <param name="limit">Total number of members to request. This must be greater than 0 if <paramref name="query"/> is specified.</param>
+    /// <param name="presences">Whether to include the <see cref="GuildMembersChunkedEventArgs.Presences"/> associated with the fetched members.</param>
+    /// <param name="userIds">Whether to limit the request to the specified user ids. Either this or <paramref name="query"/> must not be null.</param>
+    /// <param name="nonce">The unique string to identify the response. This must be unique per-guild if multiple requests to the same guild are made.</param>
+    /// <param name="cancellationToken">A cancellation token to cancel the iterator with.</param>
+    /// <returns>An asynchronous iterator that will return all chunks.</returns>
+    public async IAsyncEnumerable<GuildMembersChunkedEventArgs> EnumerateRequestMembersAsync

```

Comment:
this should arguably return IAsyncEnumerable<DiscordMember> to abstract the whole chunking part

Reply:
It...does? Or did you mean IAsyncEnumerable<DiscordMember>

Reply:
`IAsyncEnumerable<DiscordMember>`, forgot that the generic would be interpreted as malformed html :ioa:

Reply:
Okay, Discord's markdown tomfoolery aside, I did contemplate doing that, however the chunk event contains a bunch of extra useful metadata like users that aren't found, though for the simple (and probably most common case) of "just give me every member on the server" I think that's acceptable.

Reply:
We can provide both apis imo, so advanced usecases still can use our wrapper around the raw events

Reply:
i would still like to have a "cleaner"/"more usable" API for that than just throwing the chunk at the user